specification/columns/availabilityzone.md

```diff
@@ -1,27 +1,27 @@
 # Availability Zone
 
-[Availability Zone](#glossary:availability-zone) is a provider assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance. Availability Zone is commonly used for scenarios like analyzing cross-zone data transfer usage and the corresponding cost based on where [*resources*](#glossary:resource) are deployed.
+An [*availability zone*](#glossary:availability-zone) is a provider-assigned identifier for a physically separated and isolated area within a Region that provides high availability and fault tolerance. Availability Zone is commonly used for scenarios like analyzing cross-zone data transfer usage and the corresponding cost based on where [*resources*](#glossary:resource) are deployed.
 
-The AvailabilityZone column SHOULD be present in the billing data. This column MUST be of type String and MAY contain null values.
+The AvailabilityZone column MUST be present in the billing data when the provider supports deploying resources or services within an *availability zone*. This column MUST be of type String and MAY contain null values when a charge is not specific to an *availability zone*.
```

Comment:
In that case, we have to describe this scenario of when to include and when to exclude in every column which seems unnecessary to me. I don't see an issue if a provider provides a nullable FOCUS column with nulls when it doesn't apply to them. For that reason, I'm ok without but if others feel it's needed, we can go an update all columns to specify the exclusion case. Thoughts?

Reply:
From my practitioner perspective, I would find unclear that a column is present but always populated with null values, because the provider does not support it. Indeed, I could not differentiate the case where a value is null because the provider does not support the column and the case where the provider supports the column but it's the case where the value is null. If a column is present, I would expect the provider supports it and provides non-null values in the relevant cases.

Meanwhile, specifying the exclusion cases for every column might be heavy. Maybe this could be specified in the nullable attribute? If a column is nullable, null values must not be used if the provider does not support the column: the column must be made absent.

Reply:
Totally aligned on the dataset-level presence of a column vs row-level nullability of the column. 

At the end of the day, once the datasets are combined as it's the promise of FOCUS, you're always looking at the union of columns. So while in a single provider-level dataset you _might be_ able to quickly guess if a provider might be supporting a concept/column or not, you can't easily do that once you're looking at the higher-level integrated dataset. Many providers are also now supporting creating custom projections so I'd argue the presence of a column can't always be counted on because you might only export a subset.

If you all are still aligned, I'm happy to add or you can suggest an update in the nullability attribute OR possibly in section 2 related to compliance.

Reply:
I had that thought as well... Multi-cloud or multi-dataset is still a smaller subset of overall practitioners who will use FOCUS. I see the overarching value of FOCUS being clearer expectations for practitioners whether they use one or more datasets. With that context, I would lean towards optimizing for the single dataset scenario, noting that it's irrelevant when merged with other datasets that have different levels of support.

But again, I'm looking for others to chime in. This is probably a good, quick survey we can knock out in one of the meetings.

Reply:
Indeed, the merge of FOCUS datasets raises questions.

The specification says:

> The value [of PricingUnit] MUST NOT be null if SkuPriceId is not null and MUST be null if SkuPriceId is null.

After the merge, if absent columns are handled with null values, this statement is not true for the charges (rows) of providers that do not support SkuPriceId. Another option for the merge is for a practitioner to put special values for missing columns, for example "<not supported>". Maybe there are other options.

It would be interesting to get others' view in a meeting.

Reply:
The group decided that we make the normative prose about the presence of the data in the charge record. In this case, the group decided that it should remain a SHOULD to prevent providers from going out of compliance for not providing AZ. Column presence is defined in the schema section.

Reply:
Follow up discussion topic raised as issue #367